run_tests.py

```diff
@@ -80,24 +83,47 @@ class GRREverythingTestLoader(test_lib.GRRTestLoader):
   base_class = test_lib.GRRBaseTest
 
 
+class GRROneTestLoader(test_lib.GRRTestLoader):
+  """Load one specific GRR test case."""
+  base_class = test_lib.GRRBaseTest
+
+  def loadTestsFromModule(self, _):
+    """Just return all the tests as if they were in the same module."""
+    test_cases = [
+        self.loadTestsFromTestCase(x) for x in self.base_class.classes.values()
+        if x.__name__ == self.test_class_name]
+
+    return self.suiteClass(test_cases)
+
+  def __init__(self, test_class_name):
+    self.test_class_name = test_class_name
+
+    super(GRROneTestLoader, self).__init__()
+
+
 def RunTest(test_suite, stream=None):
   out_fd = stream
   if stream:
     out_fd = StringIO.StringIO()
 
   try:
-    test_lib.GrrTestProgram(argv=[sys.argv[0], test_suite],
-                            testLoader=GRREverythingTestLoader(
-                                labels=flags.FLAGS.labels),
-                            testRunner=unittest.TextTestRunner(
-                                stream=out_fd))
+    program = test_lib.GrrTestProgram(
+                argv=[sys.argv[0], test_suite],
+                testLoader=GRROneTestLoader(
+                    test_class_name=test_suite),
+                testRunner=unittest.TextTestRunner(
+                    stream=out_fd))
+  except SystemExit as e:
+    # must include this section, as GrrTestProgram issues
```

Comment:
Hmm ok sounds good.

On 9 February 2015 at 16:20, Dionysis Zindros <notifications@github.com>
wrote:

> In run_tests.py
> <https://github.com/google/grr/pull/113#discussion_r24334668>:
>
> >  def RunTest(test_suite, stream=None):
> >    out_fd = stream
> >    if stream:
> >      out_fd = StringIO.StringIO()
> >
> >    try:
> > -    test_lib.GrrTestProgram(argv=[sys.argv[0], test_suite],
> > -                            testLoader=GRREverythingTestLoader(
> > -                                labels=flags.FLAGS.labels),
> > -                            testRunner=unittest.TextTestRunner(
> > -                                stream=out_fd))
> > +    program = test_lib.GrrTestProgram(
> > +                argv=[sys.argv[0], test_suite],
> > +                testLoader=GRROneTestLoader(
> > +                    test_class_name=test_suite),
> > +                testRunner=unittest.TextTestRunner(
> > +                    stream=out_fd))
> > +  except SystemExit as e:
> > +    # must include this section, as GrrTestProgram issues
>
> This is a work-in-progress as indicated in the "[WIP]" title. I'm in the
> process of squashing my history to something clearer so that you can review
> without trouble. I'll indicate when this is ready for review by requesting
> it with a PTAL and a removal of the WIP from the title. Thanks! :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/grr/pull/113/files#r24334668>.
>